src/aeos/config/configure.ml

```diff
@@ -5,51 +5,71 @@
 
 module C = Configurator.V1
 
+
+let igetenv v =
+  let v' = try Sys.getenv v |> int_of_string with Not_found -> 0 in
+  if v' < 0 || v' > 1 then raise @@
+    Invalid_argument (Printf.sprintf "Invalid value for env variable %s: got %d" v v');
+  v'
+
+
 let get_os_type c =
   let sys = C.ocaml_config_var c "system" in
   match sys with Some s -> s | None -> ""
 
 
-let get_default_cflags _c = [
+let default_cflags = [
   "-g"; "-O3"; "-Ofast";
   "-march=native"; "-funroll-loops"; "-ffast-math";
   "-DSFMT_MEXP=19937"; "-fno-strict-aliasing";
 ]
 
 
 let get_openmp_cflags c =
-  match get_os_type c with
-  | "linux"     -> [ "-fopenmp" ]
-  | "linux_elf" -> [ "-fopenmp" ]
-  | "macosx"    -> [ "-Xpreprocessor"; "-fopenmp" ]
-  | "mingw64"   -> [ "-fopenmp" ]
-  | _           -> []
+  let enable_openmp = igetenv "ENABLE_OPENMP" in
+  if enable_openmp = 0 then []
+  else if enable_openmp = 1 then (
+    match get_os_type c with
+    | "linux"        -> [ "-fopenmp" ]
+    | "linux_elf"    -> [ "-fopenmp" ]
+    | "linux_eabihf" -> [ "-fopenmp" ]
+    | "macosx"       -> [ "-Xpreprocessor"; "-fopenmp" ]
+    | "mingw64"      -> [ "-fopenmp" ]
+    | _              -> []
+  )
+  else failwith "Error: ENABLE_OPENMP only accepts 0/1."
 
 
-let get_default_libs () = ["-lm";]
+let default_libs = ["-lm";]
 
 
 let get_openmp_libs c =
-  match get_os_type c with
-  | "linux"     -> [ "-lgomp" ]
-  | "linux_elf" -> [ "-lgomp" ]
-  | "macosx"    -> [ "-lomp" ]
-  | "mingw64"   -> [ "-lgomp" ]
-  | _           -> []
+  let enable_openmp = igetenv "ENABLE_OPENMP" in
```

Comment:
You just need 
```ocaml
  let enable_openmp = igetenv "ENABLE_OPENMP" in
  if enable_openmp = 0 then []
  else
    match get_os_type c with
    | "linux"     -> [ "-fopenmp" ]
    | "linux_elf" -> [ "-fopenmp" ]
    | "macosx"    -> [ "-Xpreprocessor"; "-fopenmp" ]
    | "mingw64"   -> [ "-fopenmp" ]
    | _           -> []
```

It would fail in line 12 if it is not integer and not 0 or 1.

Reply:
Ops, this was meant for the block above, but it's the same here